saga-spring/src/main/java/io/servicecomb/saga/spring/SagaController.java

```diff
@@ -53,10 +59,23 @@ public SagaController(SagaExecutionComponent sagaExecutionComponent, SagaEventRe
   }
 
   @Trace("processRequests")
+  @ApiResponses({
+      @ApiResponse(code = 200, response = String.class, message = "success"),
+      @ApiResponse(code = 400, response = String.class, message = "illegal request content"),
+      @ApiResponse(code = 500, response = String.class, message = "transaction failed")
+  })
   @RequestMapping(value = "requests", method = POST, consumes = TEXT_PLAIN_VALUE, produces = TEXT_PLAIN_VALUE)
   public ResponseEntity<String> processRequests(@RequestBody String request) {
-    sagaExecutionComponent.run(request);
-    return ResponseEntity.ok("success");
+    try {
+      String runResult = sagaExecutionComponent.run(request);
+      if (runResult == null) {
+        return ResponseEntity.ok("success");
+      } else {
+        throw new InvocationException(INTERNAL_SERVER_ERROR, runResult);
+      }
+    } catch (SagaException se) {
+      throw new InvocationException(BAD_REQUEST, se.getMessage());
```

Comment:
prefer to pass exception along instead of its message only

Reply:
OK